### DISCUSSION TITLE
Error while testing code

### DISCUSSION THREAD

**Discussion by anupam198** [Cyfrin/foundry-full-course-cu#595] [General]
Whenever I m trying to test code using "Forge test -vv" command or try using "forge build" command, I am getting the below error

_Error (6275): Source "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol" not found: File not found. Searched the following locations: "/home/anupam/foundry/foundry-fund-me-f23".
 --> src/PriceConverter.sol:4:1:
  |
4 | import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";_

when infact the file "AggregatorV3Interface" is present at the desired location but ene then I am getting this error. This error is preventing me from testing and compiling my code, please help

**Comment by theirrationalone** [comment]
Check your `foundry.toml` file in there make sure to add library path in the array of remappings. 

Like this:

```toml
remappings = ['@chainlink/contracts/=lib/chainlink-brownie-contracts/contracts/']
```

**Comment by anupam198** [comment]
see my foundry.toml file

remapping has already been done even then I see this error

[profile.default]
src = "src"
out = "out"
libs = ["lib"]
remapppings = [
    "@chainlink/contracts/= lib/chainlink-brownie-contracts/contracts/",
]
# See more config options https://github.com/foundry-rs/foundry/tree/master/config

**Reply by theirrationalone** [reply]
You're naming that wrong ❌...... 

Please rename `remapppings` with 3x 'p' to `remappings` with 2x 'p'

# remappings ✅ | remapppings ❌

And please remove that extra ', ' too that is at end of the array of remappings for being super safer.

**Reply by shikhar229169** [reply]
Actually adding extra comma at the end, doesn't generate any errors. I have checked it.

**Comment by anupam198** [comment]
Thanks for helping me but when I try to save foundry.toml file the extra "," comes automatically

**Reply by theirrationalone** [reply]
# You can try to build, compile & test without removing that comma too

Extra info:

Press
Windows: `ctrl + shift + p`
Mac: `window + p`

then type save without format and find that command and then execute that feature by clicking on that.

**Comment by anupam198** [comment]
please help me with the below error a well

**_anupam@DESKTOP-L9IKQDL:~/foundry/foundry-fund-me-f23$ forge test -vv
[⠊] Compiling...
Error: 
Failed to resolve file: "/home/anupam/foundry/foundry-fund-me-f23/ lib/chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol": No such file or directory (os error 2).
 Check configured remappings..
    --> "/home/anupam/foundry/foundry-fund-me-f23/src/FundMe.sol"
        "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol"_**

**Reply by theirrationalone** [reply]
Please remove that extra space bar too from remappings.. Before lib. 

# = lib/ wrong ❌
# =lib/ correct✅

**Reply by shikhar229169** [reply]
Actually in your remappings there is an extra space after = symbol, so remove that.
```
remapppings = ["@chainlink/contracts/=lib/chainlink-brownie-contracts/contracts/",]
```

**Reply by shikhar229169** [reply]
Woah @theirrationalone, lol we added at the same exact time.

**Reply by theirrationalone** [reply]
We are enough buddy 😘😘😂😂

**Reply by anupam198** [reply]
Thanks a ton both

**Reply by anupam198** [reply]
one new issue that I am encountering

_**forge test -m testPriceFeedVerrsionIsAccurate -vvv --fork-url $SEPOLIA_RPC_URL

error: unexpected argument '-m' found

Usage: forge test [OPTIONS]

For more information, try '--help'.**_

**Reply by theirrationalone** [reply]
`-m` outdated and updated with command `--mt` !

**Comment by anupam198** [comment]
I keep asking you about issues I am facing while testing this

forge test --mt testPriceFeedversionIsAccurate -vv --fork-url$SEPOLIA_RPC_URL

**_error: a value is required for '--fork-url <URL>' but none was supplied

For more information, try '--help'._**

I have already set up the env file with sepolia but even then I am getting the above error

**Reply by theirrationalone** [reply]
Make sure to execute `source .env` into terminal before running your Tests and one more thing...

there you need to have a `White Space | Space Bar` in between `--fork-url` & `$SEPOLIA_RPC_URL` like this below....
```terminal
 --fork-url $SEPOLIA_RPC_URL
```

**Reply by anupam198** [reply]
getting the same error in below command

**_forge test -mt testFundFailsWithoutEnoughEth_**
error: unexpected argument '-m' found

Usage: forge test [OPTIONS]

For more information, try '--help'.

**Reply by theirrationalone** [reply]
that two times - i.e. `--` not one time `-` ! 😉
